### DISCUSSION TITLE
Accessing 'CONTACT_LAST_UPDATED_TIMESTAMP' constant or 'DIRTY' flag

### DISCUSSION THREAD

**Discussion by lati111** [vestrel00/contacts-android#353] [Ideas]
I'm wanting to move local changes to a contact on a user's device to the remote database during syncing. Unfortunately as for what I can see in the documentation there does not seem to be any indicator of whether the user has edited the synced contact. 

Android does have built in support for this within their contacts, using the [CONTACT_LAST_UPDATED_TIMESTAMP](https://developer.android.com/reference/android/provider/ContactsContract.ContactsColumns#CONTACT_LAST_UPDATED_TIMESTAMP) constant, or the [DIRTY](https://developer.android.com/reference/android/provider/ContactsContract.RawContacts#columns) flag on a rawContact. 

If it all possible, I'd like it if one of these two options would be made available to allow me to determine if the item was updated on the remote database, or locally in order to set the sync direction. 

Thanks in advance.

**Comment by vestrel00** [comment]
Hey! It seems like you are implementing  a sync adapter. 

The [CONTACT_LAST_UPDATED_TIMESTAMP](https://developer.android.com/reference/android/provider/ContactsContract.ContactsColumns#CONTACT_LAST_UPDATED_TIMESTAMP) is for Contacts, which may consist of several RawContacts from different Accounts. This is probably not what you want.

I believe what you need here is [DIRTY](https://developer.android.com/reference/android/provider/ContactsContract.RawContacts#columns)

![Screenshot 2024-07-19 at 6 32 53 AM](https://github.com/user-attachments/assets/93e2bcef-a742-4f57-b4e6-b5c5b7c2e631)

**Comment by vestrel00** [comment]
@lati111, I created these issues for you;

- #354 
- #355 

They will be part of the next release [v0.3.3](https://github.com/vestrel00/contacts-android/discussions/349), which I'm aiming to release some time this September. I currently have no time to work on this until then.

In the meantime, I think you can still use this library but you'll have to write your own custom function that uses `ContactsContract` directly to read/write the value of the `DIRTY` column 🙏

**Reply by lati111** [reply]
Much appreciated! And that was the way I ended up implementing it anyways for now, even if it's a bit clunky execution wise. I'll just have to refactor once the update comes out. As always thanks for the quick support!
